sql-math-functions (#2228)
Codecademy/docs: content/sql/concepts/math-functions/math-functions.md

---
Title: 'Math Functions'

> Description: 'SQL Math Functions are built-in features in SQL, which allow mathematical operations directly in database queries.'

Reviewer: ```suggestion
Description: 'In SQL, math functions are built-in features which perform mathematical operations and complex calculations directly in database queries.'
```

Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'SQLite'
  - 'MySQL'
  - 'PostgreSQL'
CatalogContent:
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> In SQL, **Math Functions** are built-in features which perform mathematical operations and complex calculations directly in database queries. This makes it possible to work with numerical data in SQL rather than extracting the data and processing it separately.

Reviewer: ```suggestion
In SQL, **math functions** are built-in features which perform mathematical operations and complex calculations directly in database queries. This makes it possible to work with numerical data in SQL rather than extracting the data and processing it separately.
```

---
Title: 'Math Functions'

> Description: 'SQL Math Functions are built-in features in SQL, which let you do mathematical operations, directly in your database queries.'

Reviewer 2: ```suggestion
Description: 'SQL Math Functions are built-in features in SQL, which allow mathematical operations directly in database queries.'
```
We don't use first- or second-person pronouns in Docs.

Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'SQLite'
  - 'MySQL'
  - 'PostgreSQL'
CatalogContent:
  - 'learn-sql'
  - 'paths/analyze-data-with-sql'
---

> **SQL Math Functions** are built-in features in SQL, which allow mathematical operations and complex calculations, directly in database queries. This makes it easier and more efficient to work with numerical data in SQL, as there's no need to extract the data and process it separately.

Reviewer: ```suggestion
In SQL, **Math Functions** are built-in features which perform mathematical operations and complex calculations directly in database queries. This makes it possible to work with numerical data in SQL rather than extracting the data and processing it separately.
```

Reviewer: Can you match the metadata's description to this introduction? TY

> 

Reviewer: You can add a sentence here that will preface the table of terms. So in conclusion, this entry will only include this introductory paragraph above. And a line below it (please add a new line in between) that says something like "Below is a list math functions that can be used in various SQL platforms:"

cc: @SSwiniarski would platforms be the term we want to use here to describe the different flavors/implementations of SQL?

Reviewer 2: @yangc95, I think that works.

> ### Example

Reviewer: For this concept entry, let's steer away from using examples that would appear as a term entry.

In response to: https://github.com/Codecademy/docs/issues/2218#issuecomment-1559350926

Take this entire example section and move it to that term entry you will create.

> **SQL Math Functions** are built-in features in SQL, which let you do mathematical operations, such as addition, subtraction, and even more complex calculations, directly in your database queries. This makes it easier and more efficient to work with numerical data in SQL, as you don't need to extract the data and process it separately.

Reviewer 2: ```suggestion
**SQL Math Functions** are built-in features in SQL, which allow mathematical operations and complex calculations, directly in database queries. This makes it easier and more efficient to work with numerical data in SQL, as there's no need to extract the data and process it separately.
```
No first- or second-person pronouns in Docs.

Also, multiplication, addition, subtraction and division are operators, not functions.

> In this example, the following data is given, in a table named `employees`:

Reviewer: ```suggestion
In this example, the following data is given in a table named `employees`:
```

| order_id | product_price | quantity |
| -------- | ------------- | -------- |
| 1        | 20            | 3        |
| 2        | 15            | 5        |
| 3        | 30            | 2        |
| 4        | 10            | 1        |

> The multiplication (\*) math function can be used to calculate the total price for each order:

Reviewer 2: Multiplication (and addition, subtraction and division) is an operator, we'd like to see an example with an actual function. such as `ABS()`, `FLOOR()` or `CELING()`

```sql
SELECT order_id, product_price, quantity, product_price * quantity AS total_price
FROM orders;
```

In this SQL statement, `product_price` and `quantity` are multiplied for each order to calculate `total_price`. The [`AS`](https://www.codecademy.com/resources/docs/sql/commands/as) keyword is used to rename the `product_price * quantity` column to `total_price` in the output.

The output will be:

| order_id | product_price | quantity | total_price |
| -------- | ------------- | -------- | ----------- |
| 1        | 20            | 3        | 60          |
| 2        | 15            | 5        | 75          |
| 3        | 30            | 2        | 60          |
| 4        | 10            | 1        | 10          |

> This is a simple but practical example of math functions in SQL.

Reviewer 2: ```suggestion
This is a practical example of math functions in SQL.
```
From the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#writing-entries):
>- Avoid words or phrases that assume the reader’s knowledge such as "easy", "obviously", "simply", "of course", and so on.
